Docs: Update person_profiles default

Lior539 opened the pull request:
The default of person_profiles is changing from .always to .identified_only. @robbie-c already updated some of the docs, but this PR does the rest of them.

Tagging for review: 
- @robbie-c (since he did some of the previous changes)
- @marandaneto (because I always do with mobile docs 😅)

cc @abigailbramble since she requested it

bijanbwb reviewed (approved):
Looks good to me. I was going to suggest extracting the shared text into a separate reusable chunk, but I realized they're all slightly different for each language. So I think this looks good as is. 👍

marandaneto commented on contents/docs/product-analytics/_snippets/how-to-capture-anonymous-events-android.mdx:
```diff
@@ -1,16 +1,18 @@
-   apiKey = POSTHOG_API_KEY,
-   host = POSTHOG_HOST,
+    apiKey = POSTHOG_API_KEY,
+    host = POSTHOG_HOST,
 ).apply {
-   personProfiles = PersonProfiles.IDENTIFIED_ONLY
+    personProfiles = PersonProfiles.IDENTIFIED_ONLY
```
I'd rollback indentation since I keep the platforms standards

marandaneto commented on contents/docs/product-analytics/_snippets/how-to-capture-anonymous-events-ios.mdx:
````diff
@@ -1,15 +1,18 @@
 For example:
 
-```ios
+```swift
````
i think it should be `ios_swift`

Lior539 replied:
```suggestion
```ios_swift
```

marandaneto reviewed (approved):
left 2 comments otherwise LGTM

abigailbramble reviewed (changes requested):
I think there are also changes required in:

- [ ] https://github.com/PostHog/posthog.com/blob/personless-default/contents/docs/product-analytics/_snippets/how-to-capture-identified-events-backend.mdx
- [ ] https://github.com/PostHog/posthog.com/blob/personless-default/contents/docs/product-analytics/_snippets/how-to-capture-anonymous-backend-and-api.mdx
- [ ] https://github.com/PostHog/posthog.com/blob/personless-default/contents/docs/product-analytics/_snippets/how-to-capture-identified-events-android.mdx
- [ ] https://github.com/PostHog/posthog.com/blob/personless-default/contents/docs/product-analytics/_snippets/how-to-capture-identified-events-flutter.mdx
- [ ] https://github.com/PostHog/posthog.com/blob/personless-default/contents/docs/product-analytics/_snippets/how-to-capture-identified-events-frontend.mdx
- [ ] https://github.com/PostHog/posthog.com/blob/personless-default/contents/docs/product-analytics/_snippets/how-to-capture-identified-events-ios.mdx
- [ ] https://github.com/PostHog/posthog.com/blob/personless-default/contents/docs/data/anonymous-vs-identified-events.mdx?plain=1#L157

And then also these depending on if this is now the default for backend also:
- [ ] https://github.com/PostHog/posthog.com/blob/personless-default/contents/docs/libraries/java/index.mdx?plain=1#L37
- [ ] https://github.com/PostHog/posthog.com/blob/personless-default/contents/docs/libraries/python/index.mdx?plain=1#L32
- [ ] https://github.com/PostHog/posthog.com/blob/personless-default/contents/docs/libraries/ruby/index.mdx?plain=1#L38
- [ ] https://github.com/PostHog/posthog.com/blob/personless-default/contents/docs/libraries/go/index.mdx?plain=1#L32
- [ ] https://github.com/PostHog/posthog.com/blob/personless-default/contents/docs/libraries/php/index.mdx?plain=1#L32
- [ ] https://github.com/PostHog/posthog.com/blob/personless-default/contents/docs/libraries/node/index.mdx?plain=1#L48


Should [this be updated ](https://posthog.com/docs/api/capture#anonymous-events)to let users know how to explicitly capture anonymous or identified events?

Lior539 commented:
@abigailbramble thanks for sharing the list.  Addressing your comments below. Note that identified events are the default backend, since they include a distinct id:

These are correctly flagged as still needing changes (Thank you):

- https://github.com/PostHog/posthog.com/blob/personless-default/contents/docs/data/anonymous-vs-identified-events.mdx?plain=1#L157
- Will also tweak the [API capture docs](https://posthog.com/docs/api/capture#anonymous-events) to be clearer

These changes are already in the PR:

- https://github.com/PostHog/posthog.com/blob/personless-default/contents/docs/product-analytics/_snippets/how-to-capture-identified-events-android.mdx
- https://github.com/PostHog/posthog.com/blob/personless-default/contents/docs/product-analytics/_snippets/how-to-capture-identified-events-flutter.mdx
- https://github.com/PostHog/posthog.com/blob/personless-default/contents/docs/product-analytics/_snippets/how-to-capture-identified-events-frontend.mdx
- https://github.com/PostHog/posthog.com/blob/personless-default/contents/docs/product-analytics/_snippets/how-to-capture-identified-events-ios.mdx

I've confirmed [here](https://posthog.slack.com/archives/C02E3BKC78F/p1734085599535959) that identified events are the default for backend still, so these don't need to change:
- https://github.com/PostHog/posthog.com/blob/personless-default/contents/docs/product-analytics/_snippets/how-to-capture-identified-events-backend.mdx
- https://github.com/PostHog/posthog.com/blob/personless-default/contents/docs/product-analytics/_snippets/how-to-capture-anonymous-backend-and-api.mdx
- https://github.com/PostHog/posthog.com/blob/personless-default/contents/docs/libraries/java/index.mdx?plain=1#L37
- https://github.com/PostHog/posthog.com/blob/personless-default/contents/docs/libraries/python/index.mdx?plain=1#L32
- https://github.com/PostHog/posthog.com/blob/personless-default/contents/docs/libraries/ruby/index.mdx?plain=1#L38
- https://github.com/PostHog/posthog.com/blob/personless-default/contents/docs/libraries/go/index.mdx?plain=1#L32
- https://github.com/PostHog/posthog.com/blob/personless-default/contents/docs/libraries/php/index.mdx?plain=1#L32
- https://github.com/PostHog/posthog.com/blob/personless-default/contents/docs/libraries/node/index.mdx?plain=1#L48

abigailbramble commented:
@Lior539 thanks for double checking that for the backend events!

Aha! for the changes already in the PR, I read "Identified events are captured by default". I don't know if there's something confusing in the "by default" here or if there's a change of tense needed to imply that the customer needs to do something. i.e. whether it needs to be "Identified events will be captured by default if you..." or "Events are captured as identified events if you..." or "All events are automatically identified if you..."

Completely up to you - just thinking out loud. Thanks for going through all these!

Lior539 commented:
> Aha! for the changes already in the PR, I read "Identified events are captured by default". 

Good feedback. I'll rework to avoid confusion

Lior539 commented on contents/docs/product-analytics/_snippets/how-to-capture-anonymous-events-android.mdx:
```diff
@@ -1,16 +1,18 @@
 val config = PostHogAndroidConfig(
-   apiKey = POSTHOG_API_KEY,
-   host = POSTHOG_HOST,
+    apiKey = POSTHOG_API_KEY,
+    host = POSTHOG_HOST,
```
```suggestion
   apiKey = POSTHOG_API_KEY,
   host = POSTHOG_HOST,
```

abigailbramble reviewed (approved):
Looks great - thank you ❤️
